Hmm I see a failing test in scipy in CI:

```
=================================== FAILURES ===================================
___ test_milp_timeout_16545[options0-Time limit reached. (HiGHS Status 13:] ____
[gw34] linux -- Python 3.11.8 /nix/store/gd3shnza1i50zn8zs04fa729ribr88m9-python3-3.11.8/bin/python3.11
options = {'time_limit': 0.1}, msg = 'Time limit reached. (HiGHS Status 13:'
    @pytest.mark.skipif(np.intp(0).itemsize < 8,
                        reason="Unhandled 32-bit GCC FP bug")
    @pytest.mark.slow
    @pytest.mark.parametrize(["options", "msg"], [({"time_limit": 0.1}, _msg_time),
                                                  ({"node_limit": 1}, _msg_iter)])
    def test_milp_timeout_16545(options, msg):
        # Ensure solution is not thrown away if MILP solver times out
        # -- see gh-16545
        rng = np.random.default_rng(5123833489170494244)
        A = rng.integers(0, 5, size=(100, 100))
        b_lb = np.full(100, fill_value=-np.inf)
        b_ub = np.full(100, fill_value=25)
        constraints = LinearConstraint(A, b_lb, b_ub)
        variable_lb = np.zeros(100)
        variable_ub = np.ones(100)
        variable_bounds = Bounds(variable_lb, variable_ub)
        integrality = np.ones(100)
        c_vector = -np.ones(100)
        res = milp(
            c_vector,
            integrality=integrality,
            bounds=variable_bounds,
            constraints=constraints,
            options=options,
        )
    
        assert res.message.startswith(msg)
>       assert res["x"] is not None
E       assert None is not None
lib/python3.11/site-packages/scipy/optimize/tests/test_milp.py:323: AssertionError
=============================== warnings summary ===============================
lib/python3.11/site-packages/scipy/conftest.py:9: 64 warnings
  /nix/store/m067dwmf2yslbz013vhyfn1c8yj4z1gq-python3.11-scipy-1.12.0/lib/python3.11/site-packages/scipy/conftest.py:9: UserWarning: The numpy.array_api submodule is still experimental. See NEP 47.
    import numpy.array_api
lib/python3.11/site-packages/scipy/stats/tests/test_qmc.py::TestUtils::test_geometric_discrepancy_mst_with_zero_distances
  /nix/store/m067dwmf2yslbz013vhyfn1c8yj4z1gq-python3.11-scipy-1.12.0/lib/python3.11/site-packages/scipy/stats/tests/test_qmc.py:232: UserWarning: Sample contains duplicate points.
    assert_allclose(qmc.geometric_discrepancy(sample, method='mst'), 0.5)
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED lib/python3.11/site-packages/scipy/optimize/tests/test_milp.py::test_milp_timeout_16545[options0-Time limit reached. (HiGHS Status 13:] - assert None is not None
= 1 failed, 58921 passed, 2699 skipped, 237 xfailed, 16 xpassed, 65 warnings in 541.07s (0:09:01) =
```

I wonder if that is related to the pybind update. Maybe this should be reported upstream? Or perhaps it is a sort of timeout failure that will not exist on Hydra?

--- a/pkgs/development/python-modules/pybind11/default.nix
+++ b/pkgs/development/python-modules/pybind11/default.nix
@@ -4,6 +4,8 @@
 , pythonOlder
 , fetchFromGitHub
 , cmake
+, ninja
+, setuptools
 , boost
 , eigen
 , python
@@ -37,23 +39,29 @@
     else python.stdenv;
 in buildPythonPackage rec {
   pname = "pybind11";
-  version = "2.11.1";
-  format = "setuptools";
+  version = "2.12.0";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "pybind";
-    repo = pname;
+    repo = "pybind11";
     rev = "v${version}";
-    hash = "sha256-sO/Fa+QrAKyq2EYyYMcjPrYI+bdJIrDoj6L3JHoDo3E=";
+    hash = "sha256-DVkI5NxM5uME9m3PFYVpJOOa2j+yjL6AJn76fCTv2nE=";
   };
 
   postPatch = ''
-    sed -i "/^timeout/d" pyproject.toml
+    substituteInPlace pyproject.toml \
+      --replace-fail "timeout=300" ""
   '';
 
-  nativeBuildInputs = [ cmake ];
+  build-system = [
+    cmake
+    ninja
+    setuptools
+  ];
+
   buildInputs = lib.optionals (pythonOlder "3.9") [ libxcrypt ];
-  propagatedBuildInputs = [ setupHook ];
+  propagatedNativeBuildInputs = [ setupHook ];
 
   stdenv = stdenv';
 

--- a/pkgs/development/python-modules/scipy/default.nix
+++ b/pkgs/development/python-modules/scipy/default.nix
@@ -34,8 +34,8 @@ let
   #     nix-shell maintainers/scripts/update.nix --argstr package python3.pkgs.scipy
   #
   # The update script uses sed regexes to replace them with the updated hashes.
-  version = "1.12.0";
-  srcHash = "sha256-PuiyYTgSegDTV9Kae5N68FOXT1jyJrNv9p2aFP70Z20=";
+  version = "1.13.0";
+  srcHash = "sha256-HaYk92hOREHMOXppK+Bs9DrBu9KUVUsZ0KV+isTofUo=";
   datasetsHashes = {
     ascent = "1qjp35ncrniq9rhzb14icwwykqg2208hcssznn3hz27w39615kh3";
     ecg = "1bwbjp43b7znnwha5hv6wiz3g0bhwrpqpi75s12zidxrbwvd62pj";
@@ -76,18 +76,15 @@ in buildPythonPackage {
         "doc/source/dev/contributor/meson_advanced.rst"
       ];
     })
-    (fetchpatch {
-      name = "openblas-0.3.26-compat.patch";
-      url = "https://github.com/scipy/scipy/commit/8c96a1f742335bca283aae418763aaba62c03378.patch";
-      hash = "sha256-SGoYDxwSAkr6D5/XEqHLerF4e4nmmI+PX+z+3taWAps=";
-    })
   ];
 
-  # Upstream complicated numpy version pinning is causing issues in the
-  # configurePhase, so we pass on it.
+  # Upstream says in a comment in their pyproject.toml that building against
+  # both numpy 2 and numpy 1 should work, but they seem to worry about numpy
+  # incompatibilities that we here with Nixpkgs' Python ecosystem, shouldn't
+  # experience.
   postPatch = ''
     substituteInPlace pyproject.toml \
-      --replace-fail 'numpy==' 'numpy>=' \
+      --replace-fail 'numpy>=2.0.0rc1,' 'numpy' \
   '';
 
   nativeBuildInputs = [

